pkg/operator/operator.go

```diff
@@ -412,34 +412,36 @@ func (o *Operator) cleanupOldResources(ctx context.Context) error {
 
 	// Cleanup resources without the provided annotation.
 	// Check the collector DaemonSet.
-	var ds appsv1.DaemonSet
-	if err := o.client.Get(ctx, client.ObjectKey{
+	dsKey := client.ObjectKey{
```

Comment:
Is this really necessary? Inline looks better IMO. Also extra variable tells us usually that this value is reused, but it's not, so it's bit misleading (: 

See the rationales on the style guide I wrote for Thanos project: https://github.com/thanos-io/thanos/blob/main/docs/contributing/coding-style-guide.md#avoid-defining-variables-used-only-once

Reply:
I generally agree that defining a local variable to only be used once is not ideal. My thought was that a local variable allows for less visual clutter in the function call. Because the client.ObjectKey is the second of three parameters in the Get method, the arguments end up separated over several lines of code and (I think) more difficult to read. My eyes have to go searching for the third argument after the inline definition of the second argument. As a colleague once pointed out to me, this issue would be less pronounced if the key was the final parameter instead of the middle parameter of controller-runtime. But that bell likely cannot be un-rung, at this point.

Go has made several choices, like the [omission of ternary expressions](https://go.dev/doc/faq#Does_Go_have_a_ternary_form), which intend to make code more readable. In my mind, I extend that idea to trying to trying to make sure that not too much logic is crammed into a single line of code.

Between the inline object definition and the compound conditional with a short assignment, the existing code crossed my personal threshold for complexity in a single line.